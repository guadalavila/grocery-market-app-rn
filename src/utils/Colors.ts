export const Colors = {
    primary: '#68B984',
    primaryLight: '#CFFDE1',
    greenUltraLight: '#E9F9F2',
    green: '#2DA32A',
    secondary: '#3D5656',
    yellow: '#FED049',
    orange: '#FCB753',
    white: '#FFFFFF',
    black: '#000000',
    greyLight: '#C5C5C5',
    greyUltraLight: '#F7F8FA',
    background: '#FCFDFB',
    grey: '#9f9f9f',
    divider: '#EFEEF1',
};
