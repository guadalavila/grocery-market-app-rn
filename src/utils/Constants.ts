export const STORAGE_KEY = {
    FAVORITE: '@favorite',
};
