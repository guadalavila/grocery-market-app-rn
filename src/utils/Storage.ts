import AsyncStorage from '@react-native-async-storage/async-storage';

export const storeData = async (key: string, value: string) => {
    try {
        const jsonValue = JSON.stringify(value);
        await AsyncStorage.setItem(key, jsonValue);
    } catch (e) {}
};

export const getData = async (key: string) => {
    try {
        const data = await AsyncStorage.getItem(key);
        return data ? JSON.parse(data) : undefined;
    } catch (e) {
        // error reading value
    }
};
