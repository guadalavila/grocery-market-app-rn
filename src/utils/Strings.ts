export const Strings = {
    appName: 'Grocery Market',
    hello: 'Hola!',
    welcome: 'Bienvenido!',
    viewMore: 'Ver MÃ¡s',
    search: 'Buscar...',

    cart: {
        label: 'Carrito',
        empty: 'No tenes productos en el carrito',
        checkout: 'Terminar Compra',
        add: 'Agregar al carrito',
    },

    addedProduct: 'Producto Agregado!',

    fruits: {
        label: 'Frutas',
    },
    vegetables: {
        label: 'Verduras',
    },

    favorites: {
        label: 'Mis Favoritos',
        empty: 'No tenes productos como favoritos',
    },
};
