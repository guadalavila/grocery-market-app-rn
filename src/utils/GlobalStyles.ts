import { StyleSheet } from 'react-native';
import { Colors } from './Colors';

export const GlobalStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    shadowCard: {
        shadowColor: '#000',
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 4,
    },
    row: {
        flexDirection: 'row',
    },
    justBtw: {
        justifyContent: 'space-between',
    },
    main: {
        backgroundColor: Colors.background,
    },
});

export const flex = (x: number) => {
    return {
        flex: x,
    };
};

export const Sizes = {
    XXS: 2,
    XS: 4,
    S: 8,
    M: 16,
    L: 24,
    XL: 30,
    XXL: 40,
    XXXL: 55,
};

export const FontSize = {
    XS: 8,
    S: 10,
    M: 14,
    L: 18,
    XL: 22,
    XXL: 26,
    XXXL: 32,
};
