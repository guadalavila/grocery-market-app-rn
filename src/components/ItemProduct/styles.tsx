import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { GlobalStyles, Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        backgroundColor: Colors.white,
        borderRadius: Sizes.S,
        alignContent: 'center',
        paddingHorizontal: Sizes.L,
        paddingBottom: Sizes.M,
        marginVertical: Sizes.S,
        marginHorizontal: Sizes.M,
        ...GlobalStyles.shadowCard,
    },
    title: {
        fontWeight: 'bold',
        fontSize: 16,
        color: Colors.black,
        marginBottom: Sizes.XS,
    },
    subtitle: {
        color: Colors.greyLight,
        marginBottom: Sizes.XS,
    },
    price: {
        color: Colors.black,
        fontWeight: 'bold',
        fontSize: 14,
        alignSelf: 'center',
    },
    buttonAdd: {
        backgroundColor: Colors.primary,
        borderRadius: Sizes.S,
        paddingHorizontal: Sizes.M,
        paddingVertical: Sizes.S,
    },
    image: {
        width: 100,
        height: 110,
        alignSelf: 'center',
    },
});
