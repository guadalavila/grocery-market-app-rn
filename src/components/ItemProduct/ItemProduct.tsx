import React, { useContext } from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { CartContext } from '../../context/cart-context';
import { UtilContext } from '../../context/util-context';
import { Product } from '../../models/Product';
import { Colors } from '../../utils/Colors';
import { GlobalStyles } from '../../utils/GlobalStyles';
import { Strings } from '../../utils/Strings';
import { styles } from './styles';

interface IItemProductProps {
    product: Product;
    onPress: () => void;
}

const ItemProduct = ({ product, onPress }: IItemProductProps) => {
    const { addProductToCart } = useContext(CartContext);
    const { setMessageToast, setShowToast } = useContext(UtilContext);

    const addProduct = () => {
        addProductToCart({ product, quantity: 1 });
        setShowToast(true);
        setMessageToast(Strings.addedProduct);
    };

    return (
        <TouchableOpacity testID='button' style={styles.container} activeOpacity={0.7} onPress={onPress}>
            <Image testID='image' style={styles.image} resizeMode={'contain'} source={{ uri: product.image }} />
            <Text style={styles.title}>{product.name}</Text>
            <Text style={styles.subtitle}>{product.description}</Text>
            <View style={[GlobalStyles.row, GlobalStyles.justBtw]}>
                <Text style={styles.price}>${product.price}</Text>
                <TouchableOpacity testID='button-add' style={styles.buttonAdd} activeOpacity={0.7} onPress={addProduct}>
                    <Icon testID='icon-add' name='add' size={22} color={Colors.white} />
                </TouchableOpacity>
            </View>
        </TouchableOpacity>
    );
};

export default ItemProduct;
