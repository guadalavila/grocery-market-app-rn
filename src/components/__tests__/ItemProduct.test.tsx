import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import { Colors } from '../../utils/Colors';
import ItemProduct from '../ItemProduct/ItemProduct';
describe('test ItemProduct component', () => {
    const productMock = {
        id: '6b17d300-943a-11ed-a1eb-0242ac120002',
        image: 'https://res.cloudinary.com/deoaxotzs/image/upload/v1673720991/grocery-store/banana-_yagtxh.webp',
        name: 'Banana',
        description: 'Organicas',
        price: 20.0,
    };
    const onPressMock = jest.fn();

    test('should match snapshot', () => {
        const component = render(<ItemProduct product={productMock} onPress={onPressMock} />);
        expect(component).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<ItemProduct product={productMock} onPress={onPressMock} />);
        expect(screen).toBeTruthy();
    });
    test('should render image', () => {
        render(<ItemProduct product={productMock} onPress={onPressMock} />);
        expect(screen.getByTestId('image')).toBeTruthy();
        expect(screen.getByTestId('image').props.resizeMode).toBe('contain');
    });

    test('should render Texts', () => {
        render(<ItemProduct product={productMock} onPress={onPressMock} />);
        expect(screen.getByText(productMock.name)).toBeTruthy();
        expect(screen.getByText(productMock.description)).toBeTruthy();
        expect(screen.getByText(`$${productMock.price}`)).toBeTruthy();
    });

    test('should render icon', () => {
        render(<ItemProduct product={productMock} onPress={onPressMock} />);
        expect(screen.getByTestId('icon-add')).toBeTruthy();
        expect(screen.getByTestId('icon-add').props.name).toBe('add');
        expect(screen.getByTestId('icon-add').props.size).toBe(22);
        expect(screen.getByTestId('icon-add').props.color).toBe(Colors.white);
    });

    test('should render button', () => {
        render(<ItemProduct product={productMock} onPress={onPressMock} />);
        const btn = screen.getByTestId('button');
        expect(btn).toBeTruthy();
        fireEvent.press(btn);
        expect(onPressMock).toHaveBeenCalled();
    });
});
