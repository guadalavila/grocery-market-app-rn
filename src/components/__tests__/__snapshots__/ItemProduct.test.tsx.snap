// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ItemProduct component should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 4,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingHorizontal": 24,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
  testID="button"
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://res.cloudinary.com/deoaxotzs/image/upload/v1673720991/grocery-store/banana-_yagtxh.webp",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 110,
        "width": 100,
      }
    }
    testID="image"
  />
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 4,
      }
    }
  >
    Banana
  </Text>
  <Text
    style={
      Object {
        "color": "#C5C5C5",
        "marginBottom": 4,
      }
    }
  >
    Organicas
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#000000",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      $
      20
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#68B984",
          "borderRadius": 8,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
      testID="button-add"
    >
      <MockedIoniconsIcons
        color="#FFFFFF"
        name="add"
        size={22}
        testID="icon-add"
      />
    </View>
  </View>
</View>
`;
